tests/integration/test_query.py

```diff
@@ -140,9 +140,9 @@ def get_project_config():
     try:
         relay.send_event(42)
 
-        event = mini_sentry.captured_events.get(timeout=8).get_event()
+        event = mini_sentry.captured_events.get(timeout=12).get_event()
         assert event["logentry"] == {"formatted": "Hello, World!"}
-        assert retry_count == 2
+        assert retry_count == 4
```

Comment:
I can look deeper into it but I believe it simply needs more retries because after the first 2 retries it have to wait for a bit before it gets the global config and can proceed

Reply:
Why do project configs need more retries with this PR?

Reply:
we trigger another dequeue by asking for fresh projectstates

https://github.com/getsentry/relay/pull/2697/files#diff-ed683d4a7b32e65843545cca6a8d6acd8d9e6779fb4df8cd6153bf9b8b844eb9R476